src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-page.component.ts

```diff
@@ -127,101 +133,112 @@ export class InstructorSessionResultPageComponent extends InstructorCommentsComp
+              };
+              this.sectionsModel[sectionName] = {
+                questions: [],
+                hasPopulated: false,
+                isTabExpanded: false,
+              };
+            }
+            this.isSectionsLoaded = true;
           }, (resp: ErrorMessageOutput) => {
+            this.hasSectionsLoadingFailed = true;
             this.statusMessageService.showErrorToast(resp.error.message);
           });
 
-          this.isNoResponsePanelLoaded = true;
-
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessageService.showErrorToast(resp.error.message);
-        });
+      // load question tabs
+      this.feedbackQuestionsService.getFeedbackQuestions({
+        courseId,
+        feedbackSessionName,
+        intent: Intent.INSTRUCTOR_RESULT,
+      }).subscribe((feedbackQuestions: FeedbackQuestions) => {
+        for (const question of feedbackQuestions.questions) {
+          this.questionsModel[question.feedbackQuestionId] = {
+            question,
+            responses: [],
+            statistics: '',
+            hasPopulated: false,
+            isTabExpanded: false,
+          };
+        }
+        this.isQuestionsLoaded = true;
+      }, (resp: ErrorMessageOutput) => {
+        this.hasQuestionsLoadingFailed = true;
+        this.statusMessageService.showErrorToast(resp.error.message);
+      });
 
-        // load current instructor name
-        this.instructorService.getInstructor({
-          courseId: queryParams.courseid,
-          intent: Intent.FULL_DETAIL,
-        }).subscribe((instructor: Instructor) => {
-          this.currInstructorName = instructor.name;
-        });
+      // load all students in course
+      this.studentService.getStudentsFromCourse({
+        courseId,
+      }).subscribe((allStudents: Students) => {
+        this.allStudentsInCourse = allStudents.students;
+        this.loadNoResponseStudents(courseId, feedbackSessionName);
       }, (resp: ErrorMessageOutput) => {
-        this.isFeedbackSessionLoading = false;
         this.statusMessageService.showErrorToast(resp.error.message);
       });
+
+      // load current instructor name
+      this.instructorService.getInstructor({
+        courseId,
+        intent: Intent.FULL_DETAIL,
+      }).subscribe((instructor: Instructor) => {
+        this.currInstructorName = instructor.name;
+      });
+    }, (resp: ErrorMessageOutput) => {
+      this.isFeedbackSessionLoading = false;
+      this.hasFeedbackSessionLoadingFailed = true;
+      this.statusMessageService.showErrorToast(resp.error.message);
     });
   }
 
+  loadNoResponseStudents(courseId: string, feedbackSessionName: string): void {
+    // load no response students
+    this.feedbackSessionsService.getFeedbackSessionSubmittedGiverSet({
+      courseId,
+      feedbackSessionName,
+    }).subscribe((feedbackSessionSubmittedGiverSet: FeedbackSessionSubmittedGiverSet) => {
+      // TODO team is missing
```

Comment:
What do you mean by team is missing sorry? Is there any particular reason this todo shouldn't be resolved in this PR?

Reply:
Actually, I have no idea why the TODO is there since the no response panel has team names so I thought it was meant to be some sort of marker for some feature? Maybe the senior devs can shed some light on this.

Reply:
Missing feature (I believe it is also missing in V6). If the question for a team, the no response panel will not count for that even the team a student belongs has responses.